docs/working-with-spdx.md

```diff
@@ -0,0 +1,158 @@
+# Working with SPDX
+SPDX is a popular specification for representing software bill of material (SBoM) information. Once an SBoM has been
+generated there are a number of ways to validate the contents. This guide will walk through the steps to generate an
+SBoM from a container image and validate some information from the SBoM using the example `licensechecker` plugin. By
+the end of this guide you will have:
+
+- Built a docker image
+- Generated an SBoM for that image in SPDX tag-value format
+- Pushed the SBoM to a registry
+- Configured Ratify to validate the licenses used in the docker image
+- Run the validation via the CLI
+
+For more information on additional metadata that is captured in SPDX visit the [SPDX specification docs](https://spdx.dev/specifications/).
+
+## Setup
+In order to complete this guide you will need some tools.
+
+- [docker](https://www.docker.com/get-started): This tool will be used to build the container image
+- [syft](https://github.com/anchore/syft): This tool will be used to generate an SBoM in SPDX tag-value format
+- [oras](https://github.com/oras-project/oras): This tool will be used to push the generated SBoM to the registry
+- [ratify](https://github.com/deislabs/ratify): This tool will be used to validate the SBoM
+
+You will also need a registry to push your container image and SBoM to. For this guide deploy a local registry with
+oras artifacts support:
+```shell
+docker run --name sbom_demo -d -p 5000:5000 ghcr.io/oras-project/registry:v0.0.3-alpha
+```
+
+## Build the Image
+First we need to build the container image we will be using:
+```shell
+docker build -t localhost:5000/only-spdx:v1 https://github.com/wabbit-networks/net-monitor.git\#main
+```
+
+## Generate the SBoM
+Generate an SPDX SBoM using syft:
+```shell
+syft -o spdx --file sbom.spdx localhost:5000/only-spdx:v1
+```
+
+## Push the image and SBoM
+After building the image and generating the SBoM push them to the registry:
+```shell
+docker push localhost:5000/only-spdx:v1
+
+oras push localhost:5000/only-spdx \
+  --artifact-type application/vnd.ratify.spdx.v0 \
+  --subject localhost:5000/only-spdx:v1 \
+  --plain-http \
+  sbom.spdx:application/text
+```
+
+## Validating the SBoM
+If you have not done so already, build and install ratify:
+```shell
+make build
+make install
+```
+
```

Comment:
where do you run these commands?

Reply:
Updated docs to indicate this should be run in the root of the cloned ratify repository.